tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: 'How to use CPM to generate an off-chain TypeScript SDK'
+description: "The tutorial aims to provide a comprehensive guide for developers interested in creating TypeScript SDKs for Neo smart contracts, making it easier for them to leverage CPM and successfully work with off-chain SDKs."
+author: coz.io
+tags: ["TS", "CPM", "SDK"]
+skill: BEGINNER
+sidebar: true
+---
+
+[CPM](https://github.com/CityOfZion/cpm) is a tool that can help you download smart contracts and generate on-chain and off-chain SDKs for multiple languages without the need of writing them yourself and needing none or minimal changes.
+## 1. Prerequisites
+You need to have [CPM installed](https://github.com/CityOfZion/cpm/tree/master#installation), and your TypeScript project will need to have the [`@types/node`](https://www.npmjs.com/package/@types/node) and [`@cityofzion/neon-dappkit-types`](https://www.npmjs.com/package/@cityofzion/neon-dappkit-types) dependencies, and also some class that implements the `Neo3-Invoker` class, like [`WalletConnect SDK`](https://github.com/CityOfZion/wallet-connect-sdk/tree/main) or [`NeonInvoker`](https://github.com/CityOfZion/neon-dappkit/blob/main/packages/neon-dappkit/src/NeonInvoker.ts) from [`@cityofzion/neon-dappkit`](https://www.npmjs.com/package/@cityofzion/neon-dappkit).
+
+## 2. Using CPM 
+It's pretty easy to use CPM to add an off-chain SDK in your TypeScript project. After installing CPM, you can run it with a `cpm.yaml` or you could directly use the command prompt. In this tutorial let's suppose we want to easily invoke functions from the [Neo Token contract](https://dora.coz.io/contract/neo3/mainnet/0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5).
+### With `.yaml` file
+You can generate a default `cpm.yaml` file by running `cpm init` on the terminal, it should create a file that is getting the [Props](https://github.com/CityOfZion/props) smart contract. Let's suppose the contract you want to create a SDK of is the NEO token contract, after tweaking the `cpm.yaml` file it should be something like this:
+
+```yaml
+defaults:
+  contract-source-network: mainnet
+  contract-generate-sdk: true
+  contract-download: false
+  off-chain:
+    languages:
+    - ts
+
+contracts:
+  - label: NEO token
+    script-hash: '0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5'
+
+networks:
+  - label: mainnet
+    hosts:
+      - 'https://mainnet1.neo.coz.io:443'
+      - 'http://seed1.neo.org:10332'
+  - label: testnet
+    hosts:
+      - 'https://testnet1.neo.coz.io:443'
+  - label: priv
+    hosts:
+      - 'http://127.0.0.1:10332'
+
+```
+
+Then just run `cpm run` on the terminal and the SDK should be generated at the `cpm/offchain/ts/neotoken` folder. Check CPM's [documentation](https://github.com/CityOfZion/cpm/blob/master/docs/config.md) to see other options and how to better configure this `cpm.yaml` file.
```

Comment:
Let's split this text in 2 parts, and since we are not asking to the reader to edit the yaml file anymore, we can simply make a small observation that he can also edit the file following the guide.